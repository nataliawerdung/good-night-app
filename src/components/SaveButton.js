import React, { Component } from 'react'
import styled from 'react-emotion'

const StyledSaveButton = styled('section')`
  display: flex;
  align-items: center;
  background: coral;
  border: 1px solid skyblue;
  border-radius: 4px;
  grid-area: save;
`

export default class SaveButton extends Component {
  render() {
    return (
      <StyledSaveButton>
        <button onClick={this.props.onClick}>save</button>
      </StyledSaveButton>
    )
  }
}
