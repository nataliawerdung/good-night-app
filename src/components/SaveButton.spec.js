import SaveButton from './SaveButton'
import { shallow } from 'enzyme'
import React from 'react'
import toJson from 'enzyme-to-json'

describe('SaveButton', () => {
  let wrapper

  let mockFunction = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<SaveButton onClick={mockFunction} />)
  })

  it('renders a SaveButton', () => {
    expect(wrapper.html()).toContain('save')
  })

  it('has a matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('reacts on click', () => {
    expect(mockFunction.mock.calls.length).toBe(0)
    wrapper.find('button').simulate('click')
    expect(mockFunction.mock.calls.length).toBe(1)
  })
})
