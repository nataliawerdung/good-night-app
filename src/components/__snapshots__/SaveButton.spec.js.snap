// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButton has a matching snapshot 1`] = `
<Styled(section)>
  <button
    onClick={[Function]}
  >
    save
  </button>
</Styled(section)>
`;
